### PR TITLE
repl: Add a wasm LibJS REPL

### PR DESCRIPTION
**Pull request by alimpfard**
This can be merged independently of the serenity changes for now, as it's just a snapshot of `master`.

image

image

### CODE REVIEW

File: repl/index.html [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    <noscript
+      ><img
+        src="https://shynet.linus.dev/ingress/441b8324-3bdc-423d-b32b-5443edb43a26/pixel.gif"
+    /></noscript>
+    <script
+      defer
+      src="https://shynet.linus.dev/ingress/441b8324-3bdc-423d-b32b-5443edb43a26/script.js"
+    ></script>
+    <script src="https://unpkg.com/unfetch/polyfill"></script>
+    <script src="/test262/fetch.js"></script>
```
Comment (linusg) [inline repl/index.html:87]: This only defines `fetchData`, why is it needed?

File: repl/index.html [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        href="https://twitter.com/linusgroh"
+        target="_blank"
+        rel="noopener noreferrer"
+        >@linusgroh</a
+      >
+      &amp;
+      <a href="https://nwex.de" target="_blank" rel="noopener noreferrer"
+        >networkException</a
+      >
+      &amp; davidot.
```
Comment (linusg) [inline repl/index.html:59]:
```suggestion
      .
```

File: repl/index.html [hunk 3]
```diff
@@ -0,0 +1,92 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>LibJS REPL</title>
+    <link rel="shortcut icon" type="image/png" href="/favicon.png" />
+    <link rel="stylesheet" href="main.css" />
+    <link rel="stylesheet" href="/test262/main.css" />
```
Comment (linusg) [inline repl/index.html:9]: _probably_ not needed, otherwise we should extract the common styles

File: repl/libjs.js [hunk 4]
```diff
@@ -0,0 +1,4879 @@
...
+    var http = new XMLHttpRequest();
+    out('Dumping out file "' + filename + '" with ' + data.length + ' bytes of data.');
+    http.open("POST", "stdio.html?file=" + filename, true);
+    http.send(data); // XXX  this does not work in workers, for some odd reason (issue #2681)
+  };
+
+  if (typeof document != 'undefined') {
+    emrun_register_handlers();
+  }
+}
```
Comment (linusg) [inline repl/libjs.js:4879]: bit sad having to commit all of this into git, we really need a build step for this website

File: repl/repl.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
+if (typeof Module === "undefined")
+  throw new Error("LibJS.js must be loaded before repl.js");
```
Comment (linusg) [inline repl/repl.js:2]: wait, since when does prettier add a line break here without curly braces? :exploding_head:

**Review by linusg — COMMENTED** [review]
Awesome! Visually a bit barebones but we can touch that up later :)

File: repl/index.html (hunk 1 above)
Comment (alimpfard) [reply repl/index.html:87]: It's probably not, I blindly copied the whole block and forgot to remove the useless ones

File: repl/index.html (hunk 3 above)
Comment (alimpfard) [reply repl/index.html:9]: It uses the colour variables and main page styles still.

File: repl/libjs.js (hunk 4 above)
Comment (alimpfard) [reply repl/libjs.js:4879]: What would be the alternative to committing it here? Even if we build it when we update the tests, wouldn't we end up committing them here again?

File: repl/repl.js (hunk 5 above)
Comment (alimpfard) [reply repl/repl.js:2]: 🤯 I didn't even notice...

**Comment by alimpfard** [comment]
> Visually a bit barebones

I thought a lot about what it should have, I just couldn't come up with anything interesting to add
Options for bytecode/optimise/whatever would be nice, but those are usually hidden on the page to avoid clutter...

File: repl/index.html (hunk 3 above)
Comment (alimpfard) [reply repl/index.html:9]: The "shared" rules take up more than 80% of the whole file, sounds fine to just pull in like this, but I can try to split them into a separate file if you want

File: repl/libjs.js (hunk 4 above)
Comment (linusg) [reply repl/libjs.js:4879]: yeh, I forgot how much github pages sucks in that regard. My plan is still to just move it to netlify or something, but this is fine for now.

File: repl/index.html (hunk 3 above)
Comment (linusg) [reply repl/index.html:9]: nah, it's fine :)

File: repl/libjs.js (hunk 4 above)
Comment (alimpfard) [reply repl/libjs.js:4879]: :thinking: we _could_ make them available as tagged "releases" in some repo and download them straight from github?

**Comment by linusg** [comment]
> I thought a lot about what it should have, I just couldn't come up with anything interesting to add

I really just meant visually, nothing wrong with the functionality

The dashed borders look a bit odd, and it could probably use some padding here and there, and maybe a different background for the boxes vs the main page... but details :)

**Review by linusg — APPROVED** [review]
let's dew it!

**Pull request merged**